linq4j/src/test/java/org/apache/calcite/linq4j/test/InlinerTest.java

```diff
@@ -152,6 +154,31 @@ void checkAssignInConditionOptimizedOut(int modifiers, String s) {
     assertThat(Expressions.toString(builder.toBlock()), is(s));
   }
 
+  /**
+   * Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6109">[CALCITE-6109]
+   * OptimizeShuttle should not create new instances of TernaryExpression
+   * if it does not do any optimization</a>.
+   */
+  @Test void testInlineTernaryNonOptimized() {
+    Statement originStatement =
+        Expressions.return_(null,
+            Expressions.makeTernary(ExpressionType.Conditional,
+                Expressions.makeBinary(ExpressionType.NotEqual,
+                    Expressions.parameter(int.class, "a"),
+                    Expressions.constant(1)),
+                Expressions.parameter(int.class, "b"),
+                Expressions.parameter(int.class, "c")));
+    b.add(originStatement);
+    BlockStatement block = b.toBlock();
+    assertEquals(block.statements.size(), 1);
```

Comment:
use `assertThat` rather than `assertEquals` and `assertSame`. people get the args in the wrong order. as you did.

Reply:
Do you mean that assertSame and assertEquals easily make people confused which is expected and which is actual? So I need to change to assertThat, right?